src/plugin/pagebreaks.js

```diff
@@ -4,15 +4,46 @@ var orig = {
   toContainer: Worker.prototype.toContainer
 };
 
+/**
+ * html2pdf__page-break-before - Page break Before the Element the pagebreak is on.
+ * html2pdf__page-break-after - Page break After the Element the pagebreak is on.
+ * html2pdf__page-break - Same behaviour as html2pdf__page-break-after.
+ * html2pdf__page-break-auto - Automatically determine if the page break is necessary,
+ *                             this is usefull on elements like images that you
+ *                             do not want to break.
+ */
 Worker.prototype.toContainer = function toContainer() {
   return orig.toContainer.call(this).then(function toContainer_pagebreak() {
-    // Enable page-breaks.
+    var pageBreaksBefore = this.prop.container.querySelectorAll('.html2pdf__page-break-before');
+    var pageBreaksAfter = this.prop.container.querySelectorAll('.html2pdf__page-break-after');
     var pageBreaks = this.prop.container.querySelectorAll('.html2pdf__page-break');
+    var pageBreakAvoid = this.prop.container.querySelectorAll('.html2pdf__page-break-auto');
     var pxPageHeight = this.prop.pageSize.inner.px.height;
+    function toPx(val, k) {
+      return Math.floor(val * k / 72 * 96);
+    }
+    Array.prototype.forEach.call(pageBreakAvoid, function pageBreak_loop(el) {
+      el.style.display = 'block';
+      var clientRect = el.getBoundingClientRect();
+      var margin = toPx(this.opt.margin[0], this.prop.pageSize.k);
```

Comment:
Since `pxPageHeight` uses `pageSize.inner` (the "inner" size), none of the other calculations should need to account for `margin`.